artisatomic/readmonsdata.py

```diff
@@ -0,0 +1,198 @@
+            datafile_transitions, unpack=True, delimiter=","
+        )
+    artisatomic.log_and_print(flog, f"transitions: {len(energy_levels_lower_1000percm)}")
+
+    energy_levels_lower_percm = energy_levels_lower_1000percm * 1000
+
+    # Get index of lower level of transition
+    lowerlevel = np.array(
+        [
+            (
+                np.abs(
+                    energiesabovegsinpercm - energylevellower
+                )  # get closest energy in energy level array to lower level
+            ).argmin()  # then get the index with argmin()
+            for energylevellower in energy_levels_lower_percm
+        ]
+    )
+
+    ionization_energy_in_ev_nist = artisatomic.get_nist_ionization_energies_ev()[(atomic_number, ion_stage)]
+
+    # get energy of upper level of transition
+    energy_levels_lower_ev = energy_levels_lower_percm * hc_in_ev_cm
+    transitionenergyev = hc_in_ev_angstrom / transition_wavelength_A
+    ionization_energy_in_ev = max(transitionenergyev)
+    artisatomic.log_and_print(
+        flog, f"ionization energy: {ionization_energy_in_ev} eV (NIST: {ionization_energy_in_ev_nist} eV)"
+    )
+
+    # If ionisation potential in data does not match NIST to within 1 decimal place
+    # then use NIST instead (probably more accurate?)
+    if round(ionization_energy_in_ev, 1) != round(ionization_energy_in_ev_nist, 1):
+        ionization_energy_in_ev = ionization_energy_in_ev_nist
+        artisatomic.log_and_print(
+            flog, f"Energies do not match -- using NIST value of {ionization_energy_in_ev_nist} eV"
+        )
+
+    energy_levels_upper_ev = transitionenergyev + energy_levels_lower_ev
+    energy_levels_upper_percm = energy_levels_upper_ev / hc_in_ev_cm
+
+    # Get index of upper level of transition
+    upperlevel = np.array(
+        [
+            (
+                np.abs(energiesabovegsinpercm - energylevelupper)  # get closest energy in energy level array
+            ).argmin()  # then get the index with argmin()
+            for energylevelupper in energy_levels_upper_percm
+        ]
+    )
+
+    # Get A value from oscillator strength
+    A_ul = np.array(
+        [
+            (
+                (8 * np.pi**2 * const.e.value**2)
+                / (
+                    const.m_e.value * const.c.value * (transition_wavelength_A[transitionnumber] / 1e10) ** 2
+                )  # convert wavelength from angstrom to m
+                * (g_arr[lower] / g_arr[upper])
+                * oscillator_strength[transitionnumber]
+            )
+            for transitionnumber, (lower, upper) in enumerate(zip(lowerlevel, upperlevel, strict=False))
+        ]
+    )
+
+    transitions = [
+        TransitionTuple(
+            lowerlevel=lowerlevel[transitionnumber],
+            upperlevel=upperlevel[transitionnumber],
+            A=A_ul[transitionnumber],
+            coll_str=-1,
+        )
+        for transitionnumber, _ in enumerate(lowerlevel)
```

Comment:
if you're not using the lowerlevel value from enumerate, why use enumerate? Better to zip lowerlevel, upperlevel, A_ul with strict=True to make sure they match lengths.